src/lwc/geTemplateBuilderFormField/geTemplateBuilderFormField.js

```diff
@@ -31,7 +35,43 @@ export default class geTemplateBuilderFormField extends LightningElement {
     @wire(getObjectInfo, { objectApiName: '$targetObjectApiName' })
     wiredObjectInfo(response) {
         if (response.data) {
-            this.objectDescribeInfo = response.data;
+            this.targetObjectDescribeInfo = response.data;
+            this.validate();
+        }
+    }
+
+    renderedCallback() {
+        if (!this.hasRendered) {
+            this.hasRendered = true;
+
+            if (this.objectApiName === DI_BATCH_INFO.objectApiName) {
+                return;
+            }
+
+            if (isEmpty(this.fieldMapping)) {
+                const inputField = this.template.querySelector('[data-id="formField"]');
+
+                inputField.setCustomValidity('Field not found');
+                inputField.reportValidity();
+
+                this.dispatchEvent(new CustomEvent(FIELD_METADATA_ERROR));
+            }
+        }
+    }
+
+    /*******************************************************************************
+     * @description Performs form field-level validations that require data not present in the parent component
+     * such as target object describe metadata.
+     */
+    validate() {
+        if (isEmpty(this.fieldMapping)) {
```

Comment:
How do we feel about one lining guard clauses like this?

`if (isEmpty(this.fieldMapping)) return;`

Reply:
I think we technically want to always drop the curly braces in around conditional clauses according to our style guid, but on a personal level this doesn't bother me one bit.

Reply:
I'd prefer sticking with the style guide if we're okay with that, especially if that's the guide we use whenever we move to having a linting config in the repo.